dask_expr/merge.py

```diff
@@ -0,0 +1,226 @@
+import functools
+
+from dask.dataframe.dispatch import make_meta, meta_nonempty
+from dask.utils import apply
+
+from dask_expr.expr import Blockwise, Expr
+from dask_expr.io import FromPandas
+from dask_expr.repartition import Repartition
+from dask_expr.shuffle import Shuffle, _contains_index_name
+
+
+class Merge(Expr):
+    """Abstract merge operation"""
+
+    _parameters = [
+        "left",
+        "right",
+        "how",
+        "on",
+        "left_on",
+        "right_on",
+        "left_index",
+        "right_index",
+        "suffixes",
+        "indicator",
+        "backend",
+        "shuffle_backend",
+    ]
+    _defaults = {
+        "how": "inner",
+        "on": None,
```

Comment:
I would expect `on` to not be a keyword.  I would expect it to be normalized to `left_on` and `right_on` by the user API `merge` function.  In general I'd want us to normalize all inputs we can before making an expression.

Reply:
Yes, I think we can do this. In general, I agree with the desire to rely on normalization outside of the `Expr` and to keep the logic within the `Expr` relatively "clean."  However, it is important to note that `Expr`s will **often** be initialized by other `Expr`s within `simplify`. Therefore, doing input validation in the collection API does not guarantee that an `Expr` will always get normalized inputs (there can easily be a bug in another `Expr`'s `_simplify_down`/`up` implementation.

Reply:
> However, it is important to note that Exprs will often be initialized by other Exprs within simplify

Understood.  I think that the principle here is "normalize inputs as early as possible".